### DISCUSSION TITLE
Forcing function using an Aux Variable

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#23835] [Q&A General] [answered]
for a transient heat transfer problem can I update the value of my variable in my problem domain prior to the next time step, with the value of an aux variable value at the nearest neighboring mesh and then solve the next time step with the imposed value?

**Comment by GiudGiud** [comment, marked as answer]
Hello

We typically do not allow modifying nonlinear variable values but you can do it using a MultiApp. You send the value you want to impose then send them back using a transfer. 

The cannon way to do this would be to write a UserObject that modifies the nonlinear variables. We dont have one that can do what you want afaik

Guillaume

**Reply by TLWise** [reply]
I was able to use the MultiApp to transfer between APPs. I noticed that the transferred value smears the solutions and seems to override the contribution of the original solutions.

**Reply by TLWise** [reply]
Here is a snapshot of what the solutions look like.
![FictHeatCylChn](https://user-images.githubusercontent.com/39968866/227583379-6e1b9fc1-9d9c-4771-9101-64f751fcd6e4.png)
![TrueHeatFlow](https://user-images.githubusercontent.com/39968866/227583411-61d2322b-0037-4fd7-a963-bcc271eb0f82.png)

**Reply by GiudGiud** [reply]
Well whether it smears or not depends on the algorithm in the transfer


If you use the exact same mesh and the exact same variable type (finitelement family and order) the copy transfer will not smear
